src/main/scala/io/computenode/cyfra/dsl/GMem.scala

```diff
@@ -15,26 +15,27 @@ import scala.concurrent.ExecutionContext.Implicits.*
 import scala.concurrent.{ExecutionContext, Future}
 import scala.language.postfixOps
 
-trait GMem[H <: Value]:
+trait GMem[H <: Value, R]: // should it be R <: Value?
```

Comment:
I think it should be just H here. GMem is resonsible just of storing values of type H after they are resolved from GRAM (and possibly some transfer from/to GRAM). The R should come as type param in map function, just like in a Scala List definition. 

But it is a bit tricky, as its mutable operation that needs prealocated memory. I believe we have support for it on Vulkan Compute Pipeline level, but that's another part of project. So maybe to make this MR not to big and to be gradual about your introduction, let's keep just map of T => T here, with no way to change type of resulting values in GMem.

Reply:
Still returns `Future[Array[R]]` right?